niceman/resource/session.py

```diff
@@ -267,31 +269,74 @@ def niceman_exec(self, command, args):
 
     def exists(self, path):
         """Determine if a path exists in the resource.
-        
+
         Parameters
         ----------
         path : string
             Path to check for existence in the resource environment.
-        
+
         Returns
         -------
         bool
             True if path exists
         """
         raise NotImplementedError
 
+    def _prepare_dest_path(self, src_path, dest_path,
+                           local=True, absolute_only=False):
+        """Do common handling for the destination target of `get` and `put`.
+
+        Parameters
+        ----------
+        src_path : str
+            Path to source file.
+        dest_path : str
+            Path to target file.  If `dest_path` is None, the basename is taken
+            from `src_path`.
+        local : bool, optional
+            Whether the destination is on the local machine.
+        absolute_only : bool, optional
+            Whether `dest_path` is required to be absolute.
+
+        Returns
+        -------
+        The path to the destination, possibly adjusted to add the basename.
+        """
+        if local:
+            exists = op.exists
+            mkdir = os.makedirs
+        else:
+            exists = self.exists
+            mkdir = partial(self.mkdir, parents=True)
+
```

Comment:
Interesting way to unclutter your if statements below by setting up your function calls here. Will remember the partial() trick.